### DISCUSSION TITLE
Question about StandardScaler and prediction

### DISCUSSION THREAD

**Discussion by beinathan** [online-ml/river#903] [Q&A] [answered]
Hello, thanks for your contributions to River. And I have a question I want to ask, if the model uses **StandardScaler** in preprocessing, when I use method predict_one( ), whether to get an output with a mean of 0 and a variance of 1, if so, could anyone tell me whether river has a related method that can change the predicted value back to the original scaler?

**Comment by MaxHalford** [comment, marked as answer]
Hello there. There is no method to do. You could potentially do this by accessing some internal attributes, but that's a bit hackish. May I ask what is your use case?

**Reply by beinathan** [reply]
I trained the model and tried to use it in the streaming data. When new feature x arrived, I used  method predict_one( ), but the resulting y_pred was 0 most of the time,  and the true value should be a numeric value in the range 0.01-0.2.  I wondered if it was because during training, StandardScaler changed y to a mean of 0 and a variance of 1？
Or, I'd rather ask if predict_one（） get the output---y_pred that doesn't need to be processed. Thanks!
